### ISSUE TITLE
Elasticsearch logs full of errors since 2.1.0

### ISSUE THREAD

**Issue by davmrtl** [elastic/elasticsearch#15478]
Hi,

Since I've upgraded my cluster to Elasticsearch 2.1.0, the Elasticsearch logs are full of those lines:
```
[2015-12-14 14:04:46,433][WARN ][indices.cluster          ] [ELK-211] [[syslog-2015.12.14][2]] marking and sending shard failed due to [engine failure, reason [refresh failed]]
java.lang.NullPointerException
[2015-12-14 14:04:48,309][DEBUG][action.admin.indices.stats] [ELK-211] [indices:monitor/stats] failed to execute operation for shard [[syslog-2015.12.14][2], node[1HVMXYCeTVSehoM7jTEu-g], [R], v[5], s[INITIALIZING], a[id=A1kOQfL9Timtcw9k5Y3loQ], unassigned_info[[reason=ALLOCATION_FAILED], at[2015-12-14T19:04:46.474Z], details[engine failure, reason [refresh failed], failure NullPointerException[null]]], expected_shard_size[16416442009]]
[syslog-2015.12.14][[syslog-2015.12.14][2]] BroadcastShardOperationFailedException[operation indices:monitor/stats failed]; nested: ShardNotFoundException[no such shard];
        at org.elasticsearch.action.support.broadcast.node.TransportBroadcastByNodeAction$BroadcastByNodeTransportRequestHandler.onShardOperation(TransportBroadcastByNodeAction.java:405)
        at org.elasticsearch.action.support.broadcast.node.TransportBroadcastByNodeAction$BroadcastByNodeTransportRequestHandler.messageReceived(TransportBroadcastByNodeAction.java:382)
        at org.elasticsearch.action.support.broadcast.node.TransportBroadcastByNodeAction$BroadcastByNodeTransportRequestHandler.messageReceived(TransportBroadcastByNodeAction.java:371)
        at org.elasticsearch.transport.TransportService$4.doRun(TransportService.java:350)
        at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:745)
Caused by: [syslog-2015.12.14][[syslog-2015.12.14][2]] ShardNotFoundException[no such shard]
        at org.elasticsearch.index.IndexService.shardSafe(IndexService.java:198)
        at org.elasticsearch.action.admin.indices.stats.TransportIndicesStatsAction.shardOperation(TransportIndicesStatsAction.java:98)
        at org.elasticsearch.action.admin.indices.stats.TransportIndicesStatsAction.shardOperation(TransportIndicesStatsAction.java:47)
        at org.elasticsearch.action.support.broadcast.node.TransportBroadcastByNodeAction$BroadcastByNodeTransportRequestHandler.onShardOperation(TransportBroadcastByNodeAction.java:401)
        ... 7 more
[2015-12-14 14:04:53,271][DEBUG][action.admin.indices.stats] [ELK-211] [indices:monitor/stats] failed to execute operation for shard [[syslog-2015.12.14][2], node[1HVMXYCeTVSehoM7jTEu-g], [R], v[5], s[INITIALIZING], a[id=A1kOQfL9Timtcw9k5Y3loQ], unassigned_info[[reason=ALLOCATION_FAILED], at[2015-12-14T19:04:46.474Z], details[engine failure, reason [refresh failed], failure NullPointerException[null]]], expected_shard_size[16416442009]]
[syslog-2015.12.14][[syslog-2015.12.14][2]] BroadcastShardOperationFailedException[operation indices:monitor/stats failed]; nested: IllegalIndexShardStateException[CurrentState[RECOVERING] operations only allowed when shard state is one of [POST_RECOVERY, STARTED, RELOCATED]];
        at org.elasticsearch.action.support.broadcast.node.TransportBroadcastByNodeAction$BroadcastByNodeTransportRequestHandler.onShardOperation(TransportBroadcastByNodeAction.java:405)
        at org.elasticsearch.action.support.broadcast.node.TransportBroadcastByNodeAction$BroadcastByNodeTransportRequestHandler.messageReceived(TransportBroadcastByNodeAction.java:382)
        at org.elasticsearch.action.support.broadcast.node.TransportBroadcastByNodeAction$BroadcastByNodeTransportRequestHandler.messageReceived(TransportBroadcastByNodeAction.java:371)
        at org.elasticsearch.transport.TransportService$4.doRun(TransportService.java:350)
        at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:745)
Caused by: [syslog-2015.12.14][[syslog-2015.12.14][2]] IllegalIndexShardStateException[CurrentState[RECOVERING] operations only allowed when shard state is one of [POST_RECOVERY, STARTED, RELOCATED]]
        at org.elasticsearch.index.shard.IndexShard.readAllowed(IndexShard.java:974)
        at org.elasticsearch.index.shard.IndexShard.acquireSearcher(IndexShard.java:808)
        at org.elasticsearch.index.shard.IndexShard.docStats(IndexShard.java:628)
        at org.elasticsearch.action.admin.indices.stats.CommonStats.<init>(CommonStats.java:131)
        at org.elasticsearch.action.admin.indices.stats.TransportIndicesStatsAction.shardOperation(TransportIndicesStatsAction.java:165)
        at org.elasticsearch.action.admin.indices.stats.TransportIndicesStatsAction.shardOperation(TransportIndicesStatsAction.java:47)
        at org.elasticsearch.action.support.broadcast.node.TransportBroadcastByNodeAction$BroadcastByNodeTransportRequestHandler.onShardOperation(TransportBroadcastByNodeAction.java:401)
        ... 7 more
```
It looks like there's a problem in Elasticsearch's code...

Regards,

David

**Comment by clintongormley** [comment]
Fixed by https://github.com/elastic/elasticsearch/pull/15202

**Comment by davmrtl** [comment]
When is the ETA for v2.1.1?

**Comment by jasontedor** [comment]
It's harmless debug logging, nothing is wrong. This has been addressed in #14950 to remedy the excess logging.
